shepherd/utils/storage.py

```diff
@@ -5,13 +5,15 @@
 
```

Comment:
Perhaps we can merge this file with the `MiniStorage` class?

Reply:
It certainly is a possibility. I didn't really want to introduce any more changes.

Reply:
On second thoughts, I'll go knees deep and refactor the crap out of it.